mica/stats/mag_stats/mag_stats.py

```diff
@@ -0,0 +1,790 @@
+    start = dwell['tstart']
+    stop = dwell['tstop']
+    slot = obs['slot']
+
+    # first we get slot data from mica and magnitudes from cheta and match them in time
+    # to match them in time, we assume they come in steps of 1.025 seconds, starting from the first
+    # time sample.
+    slot_data_cols = ['TIME', 'END_INTEG_TIME', 'IMGSIZE', 'IMGROW0', 'IMGCOL0', 'TEMPCCD', 'IMGRAW']
+    slot_data = aca_l0.get_slot_data(start, stop, slot=obs['slot'],
+                                     img_shape_8x8=True, columns=slot_data_cols)
+
+    msid = fetch.MSID(f'AOACMAG{slot}', start, stop)
+    if len(slot_data) == 0:
+        raise MagStatsException('No level 0 data',
+                                agasc_id=obs["agasc_id"],
+                                obsid=obs["obsid"],
+                                timeline_id=obs["timeline_id"])
+    tmin = np.min([np.min(slot_data['END_INTEG_TIME']), np.min(msid.times)])
+    t1 = np.round((msid.times - tmin)/1.025)
+    t2 = np.round((slot_data['END_INTEG_TIME'].data - tmin)/1.025)
+    c, i1, i2 = np.intersect1d(t1, t2, return_indices=True)
+    times = msid.times[i1]
+
+    # the following line removes a couple of points at the edges. I have not checked why they differ
+    slot_data = slot_data[i2]
+
+    if len(times) == 0:
+        # the intersection was null.
+        raise MagStatsException('Time mismatch between cheta and level0',
+                                agasc_id=obs["agasc_id"],
+                                obsid=obs["obsid"],
+                                timeline_id=obs["timeline_id"])
+
+    # Now that we have the times, we get the rest of the MSIDs
+    telem = {
+        'times': times
+    }
+    telem.update({k: slot_data[k] for k in slot_data_cols[2:]})
+
+    names = ['AOACASEQ', 'AOPCADMD',
+             f'AOACIIR{slot}', f'AOACISP{slot}', f'AOACMAG{slot}', f'AOACFCT{slot}',
+             f'AOACZAN{slot}', f'AOACYAN{slot}'] + [f'AOATTQT{i}' for i in range(1, 5)]
+    msids = fetch.MSIDset(names, times[0] - 4, times[-1] + 4)
+    # the following just works...
+    t = np.in1d(msids[names[0]].times, times)
+    telem.update({n: msids[n].vals[t] for n in names})
+    if len(telem['AOACASEQ']) != len(telem['IMGSIZE']):
+        raise MagStatsException(
+            "Mismatch in telemetry between aca_l0 and cheta",
+            agasc_id=obs['agasc_id'], obsid=obs['obsid'], timeline_id=obs['timeline_id']
+        )
+    for name in ['AOACIIR', 'AOACISP', 'AOACYAN', 'AOACZAN', 'AOACMAG', 'AOACFCT']:
+        telem[name] = telem[f'{name}{slot}']
+        del telem[f'{name}{slot}']
+    for name in ['AOACIIR', 'AOACISP']:
+        telem[name] = np.char.rstrip(telem[name])
+    ok = (telem['AOACASEQ'] == 'KALM') & (telem[f'AOACIIR'] == 'OK') & \
+         (telem[f'AOACISP'] == 'OK') & (telem['AOPCADMD'] == 'NPNT') & \
+         (telem['AOACFCT'] == 'TRAK')
+
+    # etc...
+    telem.update(get_mag_from_img(slot_data, start, ok))
```

Comment:
These bits aren't really consistent with the docstring for the method so might want to add some breadcrumbs up there.